File: packages/@simpli/cli-scaffold/generator/template/src/model/response/ExtensaoDoPrincipalResp.ts
/**
 * ExtensaoDoPrincipalResp
 * @author martinlabs CRUD generator
 */
import {ID, TAG, Resource} from '@/simpli'
import {ResponseSerialize} from '@/simpli'

import ExtensaoDoPrincipal from '@/model/ExtensaoDoPrincipal'
import Principal from '@/model/Principal'

export default class ExtensaoDoPrincipalResp extends Resource {
  readonly $endpoint: string = '/Crud/ExtensaoDoPrincipal{/id}'

  @ResponseSerialize(ExtensaoDoPrincipal)
  extensaoDoPrincipal: ExtensaoDoPrincipal = new ExtensaoDoPrincipal()

  @ResponseSerialize(Principal)
  allPrincipal: Principal[] = []

  get $id() {
    return this.extensaoDoPrincipal.$id
  }

  set $id(val: ID) {
    this.extensaoDoPrincipal.$id = val
  }

  get $tag() {
    return this.extensaoDoPrincipal.$tag
  }

  set $tag(val: TAG) {
    this.extensaoDoPrincipal.$tag = val
  }
}

File: packages/@simpli/cli-server/generator/template/src/main/java/br/com/martinlabs/usecase/crud/response/ExtensaoDoPrincipalResp.kt
package br.com.martinlabs.usecase.crud.response

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty
import br.com.martinlabs.usecase.model.ExtensaoDoPrincipal
import br.com.martinlabs.usecase.model.Principal

@ApiModel(value = "ExtensaoDoPrincipalResp")
class ExtensaoDoPrincipalResp {

    var extensaoDoPrincipal: ExtensaoDoPrincipal? = null

    @ApiModelProperty(value = "Possible Principal values")
    var allPrincipal: MutableList<Principal>? = null
}
